### DISCUSSION TITLE
Show which mods are incompatible with the currently selected combination

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#14304] [Ideas] [answered]
Until recently, there have been some "disabled" mods because they weren't implemented yet.
But since that doesn't seem to be the case anymore (for now), I'd like to suggest a feature that could help with the mods' UX.

(Short version:)
My idea is to disable/obscure mods that are incompatible with the currently selected combination (but still having the ability to select them) + showing with the tooltip which are the mods that make them incompatible.

---

Here's an example:

Let's assume we select `Hidden`.
`Hidden` has 3 incompatible mods:
- `Traceable`
- `Approach Different`
- `Spin In`

Currently, there's no way to check if these mods are incompatible until you select them:

https://user-images.githubusercontent.com/43073074/129458483-172d7677-5547-44aa-999b-4f7af69da3b8.mp4

Next, we want to add `Sudden Death`.
`Sudden Death` has 4 incompatible mods:
- `No Fail`
- `Relax`
- `Autopilot`
- `Autoplay`

In this case, both the `Hidden` and `Sudden Death` incompatible mods must be visually different from the compatible ones.

(**The design:**)
To tell the user that some mods are incompatible maybe they can show a red/grey tint (or simply show them semi-transparent).
Here's a very simple-but-also-ugly example with a red tint:

![ex 1](https://user-images.githubusercontent.com/43073074/129457928-4a1b235f-d850-4636-8554-7b8c9545af7f.png)

In addition, it would be interesting to know which mods have made another one incompatible:

![ex 2](https://user-images.githubusercontent.com/43073074/129457932-1df73935-5a62-4bf0-82ba-90bf0413af8d.png)

The proposal is only to visually warn of the incompatibility between mods. The user can still select them (knowing that incompatible mods will be deselected).

**Comment by hlysine** [comment]
I know a re-design of the mod select overlay is coming, but I think this is a good temporary improvement to the current design, so I've worked on it a bit:

![image](https://user-images.githubusercontent.com/25472513/130319117-815215fb-8e32-43ac-b0c3-3ae345671bc1.png)

Will PR when I have time to tidy up the code.

**Reply by bdach** [reply]
I'm not sure whether this will scale well when you have more of these incompatibilities. I think an explicit fade-out/disable like proposed in the OP is better.

**Comment by Gabixel** [comment, marked as answer]
https://github.com/ppy/osu/pull/14441
